### DISCUSSION TITLE
如何实现LayUI选项卡首页不能关闭，但其他tab选项卡能关闭

### DISCUSSION THREAD

**Discussion by Python-tool** [layui/layui#1933] [问答] [answered]
我在用layui Admin框架模板时，使用选项卡与导航栏中的选项相关联，如何实现LayUI选项卡首页不能关闭，但其他tab选项卡能关闭，类似下面图片中的首页选项卡无关闭功能。
![80a284a3301641a0a998d82f70acaa85](https://github.com/layui/layui/assets/57169212/5976e477-7ae2-44ed-b97e-9f9ed21f4198)
在网上找到一个解决方案：页面加载完成后调用“$(".layui-tab-title li").eq(0).find("i").remove();”将第一个tab的关闭按钮删除掉,然后在element.js文件中找到tabAuto: function()这个方法,再方法最后添加“$(".layui-tab-title li").eq(0).find("i").remove();”。这个方法只是打开新增TAB的话有效果，但打开了多个TAB，再关掉一个TAB时，第一个TAB又会有关闭按钮了。

**Comment by LSL1618** [comment]
在源码中找到关闭图标的点击事件代码处，根据选项卡标题判断是否允许删除选项卡。

**Reply by Python-tool** [reply]
要像上面那个图那样，首页的选项卡直接没有关闭按钮该如何实现

**Reply by LSL1618** [reply]
这样不保险，没有关闭按钮不等于无法关闭，我可以使用关闭其它选项卡按钮来关闭首页选项卡，防不住的。最保险的还是从关闭事件着手，从根源上避免。

**Reply by Python-tool** [reply]
没关系，主页没有写关闭其它选项卡按钮：
![实现样式](https://github.com/layui/layui/assets/57169212/3d958636-0cf0-4360-9ad6-2a923f85755c)
html内容：
![html](https://github.com/layui/layui/assets/57169212/fb10e14f-343e-4f96-b926-ffc62663f2b3)
js里面这样写，增加新的选项卡没问题，但是关闭一个后首页的选项卡关闭按钮又出来了：
![js](https://github.com/layui/layui/assets/57169212/afbf179c-d485-4795-9a9b-e3040ddbb777)

**Reply by LSL1618** [reply]
那你不应该使用`$(".layui-tab-title li")`去查找目标元素，最好利用首页选项卡相关的具有唯一性标识的属性，去查找目标元素并移除关闭按钮，如果能利用css自然是最好的。

**Comment by Python-tool** [comment]
求大佬指点迷津呀

**Comment by Sight-wcg** [comment, marked as answer]
目前不支持，你新开一个 issues，下个版本可能会支持

**Reply by Python-tool** [reply]
好的。换了个实现方法：不开启选贤卡关闭按钮，通过双击关闭，关闭的时候判断一下是首页的选项卡则不关闭。
![123](https://github.com/layui/layui/assets/57169212/49e433bf-f04c-4a22-9f39-f828f0d0c2f5)
